tools/codegen/api/python.py

```diff
@@ -602,6 +602,10 @@ def argument_type_str(t: Type, *, simple_type: bool = False) -> str:
         if str(t.elem) == 'Tensor':
             # Is it desired to keep '?' for simple_type with new style dispatcher?
             return 'Tensor?'
+        # int[]? resolves to a specialized OptionalIntArrayRef class
+        if isinstance(t.elem, ListType) and str(t.elem.elem) == 'int':
+            size = None if simple_type else t.elem.size
+            return f'OptionalIntArrayRef[{size}]' if size is not None else 'OptionalIntArrayRef'
```

Comment:
I don't understand why this is necessary. The values here get fed into the Python arg parser, so the stated justification for OptionalIntArrayRef (which dangling reference to temporary) doesn't apply here, because this never gets used as an actual C++ type. It's fine for us to parse out as optional<IntArrayRef<...>> and then implicitly convert to OptionalIntArrayRef when we pass onto underlying API. This would also save you from having to augment the python argument parser with more code. WDYT.

Reply:
That makes sense, and it looks like you're right! I tried undoing the changes to this file and the arg parser files, and it seems to work still